cpp/src/IceGrid/AdminSessionI.cpp

```diff
@@ -489,7 +486,10 @@ AdminSessionFactory::createGlacier2Session(const string& sessionId, const option
         }
     }
 
-    _reaper->add(make_shared<SessionReapable<AdminSessionI>>(_database->getTraceLevels()->logger, session), timeout);
+    // We can't use a non-0 timeout such as the Glacier2 session timeout. As of Ice 3.8, heartbeats may not be sent
+    // at all on a busy connection. Furthermore, as of Ice 3.8, Glacier2 no longer "converts" heartbeats into
+    // keepAlive requests.
+    _reaper->add(make_shared<SessionReapable<AdminSessionI>>(_database->getTraceLevels()->logger, session), 0s);
```

Comment:
This code is actually not correct: 0s timeout + null connection is not good, as the ReapThread will keep this wrapped servant forever.

Will fix in a follow-up PR. Same for SessionI.cpp.